prymer/primer3/primer3.py

```diff
@@ -760,3 +747,44 @@ def _create_design_region(
         )
 
         return design_region
+
+
+def _check_design_results(design_input: Primer3Input, design_results: dict[str, str]) -> int:
+    """Checks for any additional Primer3 errors and reports out the count of emitted designs."""
+    count_tag = design_input.task.count_tag
+    maybe_count: Optional[str] = design_results.get(count_tag)
+    if maybe_count is None:  # no count tag was found
+        if "PRIMER_ERROR" in design_results:
+            primer_error = design_results["PRIMER_ERROR"]
+            raise ValueError(f"Primer3 returned an error: {primer_error}")
+        else:
+            raise ValueError(f"Primer3 did not return the count tag: {count_tag}")
+    count: int = int(maybe_count)
+
+    return count
+
+
+def _has_acceptable_dinuc_run(design_input: Primer3Input, oligo_design: Oligo) -> bool:
+    """
+    True if the design's longest dinucleotide run is no more than the stipulated maximum.
+
+    For primer designs, the maximum is recorded in the input's
+    `PrimerAndAmpliconParameters.primer_max_dinuc_bases`.
+
+    For probe designs, the maximum is recorded in the input's
+    `ProbeParameters.probe_max_dinuc_bases`.
+
+    Args:
+        design_input: the Primer3Input object that wraps task-specific and design-specific params
+        oligo_design: the design candidate
+
+    Returns:
+
+    """
+    max_dinuc_bases: int
+    if design_input.task.requires_primer_amplicon_params:
+        max_dinuc_bases = design_input.primer_and_amplicon_params.primer_max_dinuc_bases
+    elif design_input.task.requires_probe_params:
+        max_dinuc_bases = design_input.probe_params.probe_max_dinuc_bases
+
+    return oligo_design.longest_dinucleotide_run_length() <= max_dinuc_bases
```

Comment:
Are these new?

Reply:
no, none of this code is a new addition. I renamed one function, and refactored existing code into `_check_design_results` so I could use it for both primer and probe design